crypto/evp_extra/evp_extra_test.cc

```diff
@@ -3061,11 +3061,14 @@ TEST_P(PerParamgenCBTest, ParamgenCallbacks) {
 
 
   // Generating an DH params will trigger the callback.
-  EVP_PKEY *pkey = EVP_PKEY_new();
   ASSERT_EQ(EVP_PKEY_paramgen_init(ctx.get()), 1);
   ASSERT_TRUE(EVP_PKEY_CTX_ctrl_str(ctx.get(), GetParam().setup_command, GetParam().setup_arg));
-  ASSERT_TRUE(EVP_PKEY_paramgen(ctx.get(), &pkey));
-  ASSERT_TRUE(pkey);
+  {
+    EVP_PKEY *pkey = EVP_PKEY_new();
+    ASSERT_TRUE(pkey);
+    ASSERT_TRUE(EVP_PKEY_paramgen(ctx.get(), &pkey));
+    EVP_PKEY_free(pkey);
+  }
```

Comment:
I agree, what is the issue?  How does putting this in a smaller scope fix it? Could this use a unique pointer to handle the cleanup?

Reply:
Static analysis thinks that the reused value is a memory leak. An easy solution was to just limit the scope of the variable, and declare another one in a different scope.

Reply:
Since the function parameter must be a pointer to a pointer, a smart pointer wouldn't work for this.

Reply:
maybe something like this would work?

```
    bssl::UniquePtr<EVP_PKEY> pkey(EVP_PKEY_new());
    ASSERT_TRUE(pkey);
    EVP_PKEY *pkey_ptr = pkey.get();
    ASSERT_TRUE(EVP_PKEY_paramgen(ctx.get(), &pkey_ptr));
```

Reply:
The proposed change would cause a double-free of the point were paramgen to fail. The logic tries to free this pointer on failure: https://github.com/aws/aws-lc/blob/0ea1cb1ac4b283b318be8e87bc02022c1215be68/crypto/fipsmodule/evp/evp_ctx.c#L580
```
int EVP_PKEY_paramgen(EVP_PKEY_CTX *ctx, EVP_PKEY **out_pkey) {
...
  if (!ctx->pmeth->paramgen(ctx, *out_pkey)) {
    EVP_PKEY_free(*out_pkey);
    *out_pkey = NULL;
    return 0;
  }
...
```